File: Patrones/Comportamiento/Strategy/Java/src/PagoConDebito.java
public class PagoConDebito implements EstrategiaDePago {
    private String email;

    public PagoConDebito(String email) {
        this.email = email;
    }

    @Override
    public void pagar(double monto) {
        System.out.println("Pagando " + monto + " usando PayPal. Email: " + email);
    }
}

File: Patrones/Comportamiento/Strategy/Python/PagoConDebito.py
from EstrategiaDePago import EstrategiaDePago


# Estrategia de pago con Débito (o PayPal)
class PagoConDebito(EstrategiaDePago):

    def __init__(self, email):
        self.__email = email

    def pagar(self, monto):
        print(f"Pagando {monto} usando PayPal. Email: {self.__email}")
